src/hotspot/cpu/x86/sharedRuntime_x86_64.cpp

```diff
@@ -3073,263 +3078,6 @@ RuntimeStub* SharedRuntime::generate_resolve_blob(address destination, const cha
   return RuntimeStub::new_runtime_stub(name, &buffer, frame_complete, frame_size_in_words, oop_maps, true);
 }
 
-#ifdef COMPILER2
```

Comment:
Again, the code below is moved to universalNativeInvoker_x86_64.cpp. And then adapted to include argument and return values shuffling as well.